app/src/main/java/org/openobservatory/ooniprobe/common/ProxySettings.java

```diff
@@ -33,8 +33,8 @@ public static ProxySettings newProxySettings(PreferenceManager pm) throws Invali
             settings.protocol = ProxyProtocol.NONE;
         } else if (protocol.equals(ProxyProtocol.PSIPHON.getProtocol())) {
             settings.protocol = ProxyProtocol.PSIPHON;
-        } else if (protocol.equals(ProxyProtocol.SOCKS5.getProtocol())) {
-            settings.protocol = ProxyProtocol.SOCKS5;
+        } else if (protocol.equals(ProxyProtocol.SOCKS5.getProtocol()) || protocol.equals(ProxyProtocol.HTTP.getProtocol()) || protocol.equals(ProxyProtocol.HTTPS.getProtocol())) {
+            settings.protocol = ProxyProtocol.valueOf(protocol.toUpperCase());
```

Comment:
Why are you using uppercase here?

Reply:
ProxyProtocol.valueOf will only accept one of the values in ProxyProtocol as in the enum definition(uppercase).